app/services/affiliate-slots.js

```diff
@@ -48,74 +86,93 @@ export default Service.extend({
   currentCountry: readOnly('geo.country'),
 
   /**
-   * Get one random unit from that can be displayed on current wiki with given `title`
+   * Get a one unit that can be displayed on search page with given `query`
+   *
+   * @param {string} query
+   * @returns {AffiliateUnit|undefined}
    */
-  getUnitOnPage(title) {
-    return this.getUnitByPriority(this.getAllUnitsOnPage(title));
+  getUnitOnSearch(query) {
+    const allUnits = this.getAllUnitsOnSearch(query);
+
+    return allUnits.length > 0 ? allUnits[0] : undefined;
   },
 
   /**
-   * Get all units that can be displayed on current wiki with given `title`
+   * Get all units that can be displayed on search page with given `query`
+   *
+   * @param {string} query
+   * @returns {AffiliateUnit[]}
    */
-  getAllUnitsOnPage(title) {
-    const availableUnitNames = this.getAvailableTargeting()
-      .filter(t => !t.disableOnPage)
-      .filter(t => checkFilter(t.page, title))
-      .map(t => t.unit)
-      .flat()
-      .filter(distinct);
-
-    return units
-      .filter(u => availableUnitNames.indexOf(u.name) > -1)
-      .filter(u => checkMobileSystem(u));
+  getAllUnitsOnSearch(query) {
+    return this.getAllUnits([
+      t => !t.disableOnSearch,
+      t => checkFilter(t.query, query),
+    ]);
   },
 
   /**
-   * Get one random unit from that can be displayed on current wiki with given `query`
+   * Get all units that can be displayed on current wiki with given `title`
```

Comment:
This description is wrong; it gets only a single unit.

Reply:
Fixed in 09a2066